### PR TITLE
Run CI on macOS using Travis CI

### PR DESCRIPTION
**Pull request by rhysd**
Currently CI is running only on Linux using Circle CI. I setup CI on macOS using Travis CI.

I checked it has successfully done: https://travis-ci.org/neovim/node-client/builds/294716293

### CODE REVIEW

**Comment by billyvg** [comment]
Looks good! Can we run both in travis and just dump circleci?

**Comment by rhysd** [comment]
I will try today!

**Comment by rhysd** [comment]
@billyvg I added Linux CI to Travis CI and removed `circleci.yml`. Could you stop Circle CI?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+after_success:
+  - codecov
+
+cache:
+  directories:
+    - $HOME/.cache/yarn
+
+notifications:
+  email:
```
Comment (billyvg) [inline .travis.yml]: Let's add the gitter webhook

**Pull request merged**